libraries/AP_Notify/Display.cpp

```diff
@@ -279,9 +280,82 @@ static const uint8_t _font[] = {
     0x00, 0x00, 0x00, 0x00, 0x00
 };
 
+#if APM_BUILD_TYPE(APM_BUILD_ArduCopter)
+static const char * _modename[] = {
+    "STAB", //   STABILIZE =     0,  // manual airframe angle with manual throttle
+    "ACRO", //   ACRO =          1,  // manual body-frame angular rate with manual throttle
+    "ALTH", //   ALT_HOLD =      2,  // manual airframe angle with automatic throttle
+    "AUTO", //   AUTO =          3,  // fully automatic waypoint control using mission commands
+    "GUID", //   GUIDED =        4,  // fully automatic fly to coordinate or fly at velocity/direction using GCS immediate commands
+    "LOIT", //   LOITER =        5,  // automatic horizontal acceleration with automatic throttle
+    "RTL ", //   RTL =           6,  // automatic return to launching point
+    "CIRC", //   CIRCLE =        7,  // automatic circular flight with automatic throttle
+    "----", //8 no mode for copter
+    "LAND", //   LAND =          9,  // automatic landing with horizontal position control
+    "----", //10 no mode for copter
+    "DRIF", //    DRIFT =        11,  // semi-automous position, yaw and throttle control
+    "----", //12 no mode for copter
+    "SPRT", //    SPORT =        13,  // manual earth-frame angular rate control with manual throttle
+    "FLIP", //    FLIP =         14,  // automatically flip the vehicle on the roll axis
+    "ATUN", //    AUTOTUNE =     15,  // automatically tune the vehicle's roll and pitch gains
+    "PHLD", //    POSHOLD =      16,  // automatic position hold with manual override, with automatic throttle
+    "BRAK", //    BRAKE =        17,  // full-brake using inertial/GPS system, no pilot input
+    "THRW", //    THROW =        18,  // throw to launch mode using inertial/GPS system, no pilot input
+    "AVOI", //    AVOID_ADSB =   19,  // automatic avoidance of obstacles in the macro scale - e.g. full-sized aircraft
+    "GNGP" //     GUIDED_NOGPS = 20,  // guided mode but only accepts attitude and altitude
+
+};
+#elif APM_BUILD_TYPE(APM_BUILD_ArduPlane)
+static const char * _modename[] = {
+    "MANU", //  = 0,
+    "CIRC", //  = 1,
+    "STAB", //  = 2,
+    "TRAN", //  = 3,
+    "ACRO", //  = 4,
+    "FBWA", //  = 5,
+    "FBWB", //  = 6,
+    "CRUS", //  = 7,
+    "ATUN", //  = 8,
+    "----", //9 no mode for plane
+    "AUTO", //  = 10,
+    "RTL ", //  = 11,
+    "LOIT", //  = 12,
+    "----", //13 no mode for plane
+    "----", //14 no mode for plane
+    "GUID", //  = 15,
+    "INIT", //  = 16,
+    "QSTB", //  = 17,
+    "QHVR", //  = 18,
+    "QLIT", //  = 19,
+    "QLND", //  = 20,
+    "QRTL" //   = 21
+};
+#else  //rover
```

Comment:
OK, in the end I've made it work somehow.
This is the commit for Copter-3.4.3 that contains (hopefully all) the code changes from above:
https://github.com/mariansoban/ardupilot/commit/fb5ee4d4c2ddb5fc92379ab02fe085292e2655e4
But it doens't work:
![dsc_0112](https://cloud.githubusercontent.com/assets/14848648/21163370/01278fe2-c193-11e6-9984-eec1155045cc.JPG)

And this is the fix:
https://github.com/mariansoban/ardupilot/commit/2c7628f3077bfdcd454edfac05e80663223eadad

![dsc_0113](https://cloud.githubusercontent.com/assets/14848648/21163397/206316c4-c193-11e6-8e50-d04909decaa6.JPG)
It's Sergey's older version of SSD1306_I2C driver I guess, I've just added there method clear_screen() to fit library requirements.
Full Copter-3.4.3 with OLED support source code is here:
https://github.com/mariansoban/ardupilot/commits/Copter-3.4-oled